dotCMS/src/main/java/com/dotcms/rendering/js/JsResource.java

```diff
@@ -550,9 +550,11 @@ private Map<Object, Object> toObjectObjectMap (final Map<String, Object> map) {
 
         final  Map<Object, Object> objectObjectMap = new HashMap<>();
 
-        for (final Map.Entry<String, Object> entry : map.entrySet()) {
+        if (Objects.nonNull(map)) {
```

Comment:
What you have should work, but for info also option to do pure streams approach like the following that removes the need to create the initial object and looping.  

return Optional.ofNullable(sourceMap)
.orElseGet(Collections::emptyMap) 
.entrySet()
.stream()
.collect(Collectors.toMap(
Map.Entry::getKey,
Map.Entry::getValue 
));